server/src/main.rs

```diff
@@ -198,15 +200,43 @@ async fn run_server(args: CliArgs) -> EdgeResult<()> {
         &edge_instance_data.clone(),
     );
 
-    let custom_headers = if let EdgeMode::Edge(edge) = &args.mode {
-        edge.custom_client_headers.clone()
-    } else {
-        vec![]
-    };
+    let (edge_info, instance_data_sender) = match &args.mode {
```

Comment:
This was previously all done in `build_caches_and_refreshers` which matched on `mode` and chose to construct the appropriate types for Edge vs Offline. Now this is done here so we know if we need to also construct a single shared Client